ble/GapAdvertisingData.h

```diff
@@ -210,9 +211,94 @@ class GapAdvertisingData
      */
     ble_error_t addData(DataType advDataType, const uint8_t *payload, uint8_t len)
     {
-        /* ToDo: Check if an AD type already exists and if the existing */
-        /*       value is exclusive or not (flags, etc.) */
+        ble_error_t result = BLE_ERROR_BUFFER_OVERFLOW;
+
+        // find field
+        uint8_t* field = findField(advDataType);
+
+        // Field type already exist, either add to field or replace
+        if (field) {
+            switch(advDataType) {
+                case FLAGS:
```

Comment:
It could be helpful to have a comment saying that the following block of types are of the replacement variety.